### PR TITLE
Add building actions on windows

### PR DESCRIPTION
**Pull request by an-tao**

### CODE REVIEW

**Comment by an-tao** [comment]
@BertrandD , Hi, would you please take a look at this PR? It fails on windows, but the part of windows is copied from drogon, I don't know why.

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -11,7 +11,47 @@ env:
...
+        working-directory: ./build
+        run: |
+          conan install .. -s compiler="Visual Studio" -s compiler.version=16 -sbuild_type=Debug -g cmake_paths
+
+      - name: Create Build Environment & Configure Cmake
+        shell: bash
+        working-directory: ./build
+        run: |
+          [[ ${{ matrix.link }} == "SHARED" ]] && shared="ON" || shared="OFF"
+          cmake .. -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=on -BUILD_TRANTOR_SHARED=$shared -DCMAKE_TOOLCHAIN_FILE="conan_paths.cmake" -DCMAKE_INSTALL_PREFIX=../install
```
Comment (BertrandD) [inline .github/workflows/cmake.yml:46]: The error message is quite strange... and with  no insight on the real reason... I was luck to find this:
```suggestion
          cmake .. -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=on -DBUILD_TRANTOR_SHARED=$shared -DCMAKE_TOOLCHAIN_FILE="conan_paths.cmake" -DCMAKE_INSTALL_PREFIX=../install
```

It works fine for the configure, but the build fails. There are some unresolved symbols. I don't have time right now to take a look at it. I'll try to find some time later in the day or in the coming days id needed.

Comment (an-tao) [reply .github/workflows/cmake.yml:46]: Thanks so much! Embarrassing mistake ; )

**Comment by an-tao** [comment]
@BertrandD Now there is a link error with c-ares.

**Comment by an-tao** [comment]
@BertrandD I tried to make a [PR](https://github.com/conan-io/conan-center-index/pull/3117) to Conan center but failed.

**Comment by BertrandD** [comment]
> @BertrandD Now there is a link error with c-ares.

I'll try to find time to fix this, but the problem might be in the  dependency build by Conan (like I had with PostgreSQL for drogon). Is it blocking for any other features ? 


> @BertrandD I tried to make a [PR](https://github.com/conan-io/conan-center-index/pull/3117) to Conan center but failed.

I never added a package to Conan center (always using private artifactory), I am not sure to be able to help. I can take a look at it. Same as for before, is it blocking any other features ?

**Comment by an-tao** [comment]
> > @BertrandD Now there is a link error with c-ares.
> 
> I'll try to find time to fix this, but the problem might be in the dependency build by Conan (like I had with PostgreSQL for drogon). Is it blocking for any other features ?
> 
> > @BertrandD I tried to make a [PR](https://github.com/conan-io/conan-center-index/pull/3117) to Conan center but failed.
> 
> I never added a package to Conan center (always using private artifactory), I am not sure to be able to help. I can take a look at it. Same as for before, is it blocking any other features ?

No, I can remove the c-ares lib from this PR

**Pull request merged**